src/Datasource/RepositoryInterface.php

```diff
@@ -158,9 +158,10 @@ public function deleteAll(Closure|array|string|null $conditions): int;
      * conditions.
      *
      * @param \Closure|array|string|null $conditions list of conditions to pass to the query
+     * @param array $options list of options to apply to the query
      * @return bool
      */
-    public function exists(Closure|array|string|null $conditions): bool;
+    public function exists(Closure|array|string|null $conditions, array $options = []): bool;
```

Comment:
Given that we replaced the $options argument for find() with a variadic $args adding adding $options for query clauses seems like a step backwards in terms of how we want our API to be going forth.

Reply:
I don't know if I understand this correcly but would you make it `mixed ...$args` here as well?

Reply:
I would rather not add anyway to specify individual query clauses.

You haven't explained your exact use case but I am pretty sure just allowing the ability to specify a custom finder would suffice. Then we can just add a 2nd $finder argument to exists().